sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -49,17 +49,22 @@ bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
     }
   }
 
-  size_t VerSize = 0;
-  Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_VERSION, 0,
-                                          nullptr, &VerSize);
+  // Some of the error handling below is special for particular OpenCL
+  // versions.  If this is an OpenCL backend, get the version.
+  const char *OpenClVer = nullptr;
+  string_class OpenClVerStr;
   if (Platform.get_backend() == cl::sycl::backend::opencl) {
-    assert(VerSize >= 10 &&
+    size_t OclVerSize = 0;
+    Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_VERSION, 0,
+                                            nullptr, &OclVerSize);
+    assert(OclVerSize >= 10 &&
            "Unexpected device version string"); // strlen("OpenCL X.Y")
+    OpenClVerStr.assign(OclVerSize, '\0');
+    Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_VERSION,
+                                            OclVerSize, &OpenClVerStr.front(),
+                                            nullptr);
+    OpenClVer = &OpenClVerStr[7]; // strlen("OpenCL ")
```

Comment:
First this comment is unclear since we do not know that it refers to 7.
`OpenClVerStr.assign(OclVerSize, '\0');`: my eyes are bleeding... :-)
I wonder whether it should be `OclVerSize + 1`.
And why a `std::string` at the first place instead of a local array?
Perhaps this kind of information should be cached in the `queue` or at least the `device`?
Perhaps the PI interface could be user-friendly?
While writing assembly code, `OpenClVer = OpenClVerStr + 7;` is simpler.
More seriously this is a motivation to move to more modern `std::string_view` in the future...

Reply:
Yes, these are good comments.  I changed the OpenCL version checking here and in the other file to use the `string` class in a more normal way.  I don't think we want to cache this information as that would spread OpenCL knowledge to even more parts of the code.  Also, this is an error handling path, so performance isn't so critical.

P.S.: Sorry about your eyes.  :-)